src/tools/compiletest/src/runtest.rs

```diff
@@ -1027,12 +1027,31 @@ impl<'test> TestCx<'test> {
     }
 
     fn set_revision_flags(&self, cmd: &mut Command) {
+        // Normalize revisions to be lowercase and replace `-`s with `_`s.
+        // Otherwise the `--cfg` flag is not valid.
+        let normalize_revision = |revision: &str| revision.to_lowercase().replace("-", "_");
+
         if let Some(revision) = self.revision {
-            // Normalize revisions to be lowercase and replace `-`s with `_`s.
-            // Otherwise the `--cfg` flag is not valid.
-            let normalized_revision = revision.to_lowercase().replace("-", "_");
+            let normalized_revision = normalize_revision(revision);
             cmd.args(&["--cfg", &normalized_revision]);
         }
+
+        if !self.props.no_auto_check_cfg {
+            let mut check_cfg = String::with_capacity(25);
+
+            // Generate `cfg(FALSE, REV1, ..., REVN)` (for all possible revisions)
+            //
+            // For compatibility reason we consider the `FALSE` cfg to be expected
+            // since it is extensively used in the testsuite.
+            check_cfg.push_str("cfg(FALSE");
```

Comment:
Do we have any documentation for this special case? If not, could you please document this behavior in the dev-guide so that the knowledge of `FALSE` being "special" exists *somewhere*?

Reply:
I'm not aware, nor was I able to find any documentation regarding `FALSE` being a de-facto cfg that will eval to false.

> could you please document this behavior in the dev-guide

Sure, I will send for that.

Reply:
rustc-dev-guide PR is up https://github.com/rust-lang/rustc-dev-guide/pull/1966